src/WebJobs.Extensions.DurableTask/Listener/TaskEntityShim.cs

```diff
@@ -48,6 +45,20 @@ public TaskEntityShim(DurableTaskExtension config, string schedulerId)
 
         public EntityId EntityId { get; private set; }
 
+        public int NumberEventsToReceive { get; set; }
+    
+        internal List<RequestMessage> OperationBatch => this.operationBatch;
+
+        public void AddOperationToBatch(RequestMessage operationMessage)
+        {
+            this.operationBatch.Add(operationMessage);
+        }
+
+        public void AddLockRequestToBatch(RequestMessage lockRequest)
+        {
+            this.lockRequest = lockRequest;
```

Comment:
>this.lockRequest = lockRequest; [](start = 12, length = 31)

I see that we check for a null lock request before calling this method, but for defense-in-depth should we throw an exception (or assert) here if `this.lockRequest` is non-null?

Reply:
Yes, that seems like a good idea.